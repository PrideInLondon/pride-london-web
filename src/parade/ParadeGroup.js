import React from 'react'
import PropTypes from 'prop-types'
import Linkedin from '../components/icons/linkedin'
import Facebook from '../components/icons/facebook'
import Twitter from '../components/icons/twitter'
import Instagram from '../components/icons/instagram'
import Globe from '../components/icons/globe'
import MailIcon from '../components/icons/mailIcon'
import Snapchat from '../components/icons/snapchat'
import Tiktok from '../components/icons/tiktok'
import theme from '../theme/theme'
import {
  RowContainer,
  CompanyContainer,
  CompanyInfoContainer,
  TitleContainer,
  IconContainer,
  CompanyDescription,
} from './ParadeGroup.styles'

const ParadeGroup = ({
  name,
  description,
  websiteUrl,
  linkedinUrl,
  twitterUrl,
  facebookUrl,
  tikTokUrl,
  instagramUrl,
  snapChatHandle,
  emailUrl,
}) => {
  return (
    <RowContainer>
      <CompanyContainer>
        <CompanyInfoContainer>
          <TitleContainer>{name}</TitleContainer>
          <CompanyDescription>
            {description ? description.description : 'No company description'}
          </CompanyDescription>
        </CompanyInfoContainer>
      </CompanyContainer>

      <IconContainer>
        {websiteUrl && (
          <a href={websiteUrl} target="_blank" rel="noopener noreferrer">
            <Globe width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {linkedinUrl && (
          <a href={linkedinUrl} target="_blank" rel="noopener noreferrer">
            <Linkedin width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {emailUrl && (
          <a href={emailUrl} target="_blank" rel="noopener noreferrer">
            <MailIcon width={30} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {facebookUrl && (
          <a href={facebookUrl} target="_blank" rel="noopener noreferrer">
            <Facebook width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {twitterUrl && (
          <a href={twitterUrl} target="_blank" rel="noopener noreferrer">
            <Twitter width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {instagramUrl && (
          <a href={instagramUrl} target="_blank" rel="noopener noreferrer">
            <Instagram width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {tikTokUrl && (
          <a href={tikTokUrl} target="_blank" rel="noopener noreferrer">
            <Tiktok width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
        {snapChatHandle && (
          <a href={snapChatHandle} target="_blank" rel="noopener noreferrer">
            <Snapchat width={24} height={24} fill={theme.colors.indigo} />
          </a>
        )}
      </IconContainer>
    </RowContainer>
  )
}

ParadeGroup.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.shape({
    description: PropTypes.string
  }),
  websiteUrl: PropTypes.string,
  twitterUrl: PropTypes.string,
  facebookUrl: PropTypes.string,
  instagramUrl: PropTypes.string,
  emailUrl: PropTypes.string,
  linkedinUrl: PropTypes.string,
  snapChatHandle: PropTypes.string,
  tikTokUrl: PropTypes.string,
}

ParadeGroup.defaultProps = {
  name: '',
  websiteUrl: null,
  twitterUrl: null,
  facebookUrl: null,
  instagramUrl: null,
  emailUrl: null,
  linkedinUrl: null,
  snapChatHandle: null,
  tikTokUrl: null,
}

export default ParadeGroup
