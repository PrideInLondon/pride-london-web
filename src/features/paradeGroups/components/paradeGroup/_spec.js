import React from 'react'
import { shallow } from 'enzyme'
import ParadeGroup from '.'

describe(ParadeGroup.name, () => {
  describe('given correct props', () => {
    it('should render as expected', () => {
      const props = {
        name: 'My parade group',
        socialInstagram: 'instagram.com/1234',
        socialFacebook: 'http://facebook.com/2j2j',
        socialTwitter: 'twitter.co/foo',
        socialWebsite: 'www.group.com',
        socialEmail: 'foo@foo.com',
      }

      const wrapper = shallow(<ParadeGroup {...props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
