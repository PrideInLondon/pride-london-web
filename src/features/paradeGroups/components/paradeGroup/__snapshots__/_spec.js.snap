// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParadeGroup given correct props should render as expected 1`] = `
<styles__RowContainer>
  <styles__TitleContainer>
    My parade group
  </styles__TitleContainer>
  <styles__IconContainer>
    <a
      href="facebook.com/2j2j"
      rel="noopener noreferrer"
      target="_blank"
    >
      <Facebook
        fill="#2D2F7F"
        height={24}
        width={24}
      />
    </a>
    <a
      href="instagram.com/1234"
      rel="noopener noreferrer"
      target="_blank"
    >
      <Instagram
        fill="#2D2F7F"
        height={24}
        width={24}
      />
    </a>
    <a
      href="twitter.co/foo"
      rel="noopener noreferrer"
      target="_blank"
    >
      <Twitter
        fill="#2D2F7F"
        height={24}
        width={24}
      />
    </a>
    <a
      href="www.group.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <Globe
        fill="#2D2F7F"
        height={24}
        width={24}
      />
    </a>
  </styles__IconContainer>
</styles__RowContainer>
`;
